packages/loader/container-loader/src/connectionStateHandler.ts

```diff
@@ -431,7 +436,9 @@ class ConnectionStateHandler implements IConnectionStateHandler {
 		// or timeout has occurred while doing so.
 		if (
 			this.pendingClientId !== this.clientId &&
-			this.hasMember(this.pendingClientId) &&
+			// shouldWaitForJoinSignal === false -> read connection AND read connections should not wait (and
+			// we will not check audience so it won't work anyway)
+			(this.hasMember(this.pendingClientId) || !this.shouldWaitForJoinSignal()) &&
```

Comment:
Could you please share a bit more on what this tries to accomplish? Is it some kind of perf optimization, a bug fix, or something else? It's hard for me to follow what is the problem that you are trying to fix

Reply:
It looks like we only transition to Connected in receivedConnectEvent() if we aren't waiting for the previous client to leave. Maybe it would make more sense not to start the wait timer in receivedAddMemberEvent() if shouldWaitForJoinSignal() === false

Reply:
Actually, I don't think what I said above will work, since we won't know at that point if we need to wait, if we don't have a connection yet